js/autotrack/common.js

```diff
@@ -57,11 +81,104 @@ const getFiberNodeComponentHierarchy = currNode => {
     builtinPropExtractorConfig
   );
 
-  return `${getFiberNodeComponentHierarchy(
+  const parentHierarchyRepresentation = getFiberNodeComponentHierarchyTraversal(
     currNode.return
-  )}${elementName};${propsString}|`;
+  );
+
+  parentHierarchyRepresentation.push({
+    elementName,
+    fiberNode: currNode,  // Needed to get all props on HeapIgnore components.
+    propsString, // :TODO: Make this an object so we can use it with 'ignoreSpecificProps'
+  });
+
+  return parentHierarchyRepresentation;
 };
 
+// Given an object array representing the component hierarchy from root to target component,
+// traverse through the element list to create the string representation of the hierarchy, taking
+// into account HeapIgnore specifications.
+// Returns an object containing:
+// * hierarchy - the string hierarchy representation of 'hierarchyArray'.
+// * heapIgnoreProps - the final set of HeapIgnore props that applies to the target component. Used
+//     to determine whether to include target text and/or ignore the entire interaction.
+const getHierarchyStringFromTraversal = hierarchyArray => {
+  let currentHeapIgnoreProps = {
+    ignoreInteraction: false,
+    ignoreInnerHierarchy: false,
+    ignoreAllProps: false,
+    ignoreTargetText: false,
+  };
+
+  // Map each hierarchy element to its string representation, considering HeapIgnore specs.
+  const hierarchyStrings = hierarchyArray.map(element => {
+    // If we're not using any part of the hierarchy (for 'ignoreInteraction') or not capturing the
+    // current subhierarchy, return an empty string for the current component.
+    if (currentHeapIgnoreProps.ignoreInteraction || currentHeapIgnoreProps.ignoreInnerHierarchy) {
+      return '';
+    }
+
+    if (currentHeapIgnoreProps.ignoreAllProps) {
+      return `${element.elementName};|`;
+    }
+
+    // Doing this at the end allows us to capture HeapIgnore components.
+    currentHeapIgnoreProps = getNewHeapIgnoreProps(currentHeapIgnoreProps, element);
```

Comment:
Why are they new?  Is there something more descriptive we could use there?

Reply:
Currently, the idea is:
* You start with some state of current HeapIgnore props
* You see an element that may or may not be a HeapIgnore component
* If it's a HeapIgnore component, you want to update the state of HeapIgnore props to take into account this HeapIgnore's props

`getUpdatedHeapIgnoreProps` would probably be more accurate